trpcultivate_phenotypes/src/Service/TripalCultivatePhenotypesTraitsService.php

```diff
@@ -381,16 +381,20 @@ public function getMethodUnit($method_id) {
     ];
 
     // Query unit.
-    $unit_id = $this->chado->query($sql, $args)
-      ->fetchField();
+    $unit_ids = $this->chado->query($sql, $args);
+    $sql = "SELECT * FROM {1:cvterm} WHERE cvterm_id = :id AND cv_id = :c_id LIMIT 1";
```

Comment:
The original issue brings up that "LIMIT 1" is very rarely a good idea. I suggest removing it here and also on the other 2 lines that were mentioned.

Reply:
Yes, you should remove the limit in all cases in an API query. Also, if your query restricts by the primary key (i.e. `FROM {1:cvterm} WHERE cvterm_id = :id`) then there should not be any other parameters (i.e. no `AND cv_id = :c_id` in the above query).

In cases where you remove the limit and the primary key is not in the where clause, then you should use `fetchAll()` or `fetchCol()` to get all the results and then confirm there is only 1.